File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/AtomicNoteEntitiesDao.java
package com.originb.inkwisenote2.modules.smartnotes.data;

import androidx.room.*;

import java.util.List;
import java.util.Set;

@Dao
public interface AtomicNoteEntitiesDao {
    @Query("SELECT * FROM atomic_note_entities WHERE note_id = :noteId")
    AtomicNoteEntity getAtomicNote(Long noteId);

    @Query("SELECT * FROM atomic_note_entities")
    List<AtomicNoteEntity> getAllAtomicNotes();

    @Query("SELECT * FROM atomic_note_entities WHERE note_id IN (:noteIds)")
    List<AtomicNoteEntity> getAtomicNotes(Set<Long> noteIds);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertAtomicNote(AtomicNoteEntity atomicNoteEntity);

    @Update
    int updateAtomicNote(AtomicNoteEntity atomicNoteEntity);

    @Update
    int updateAtomicNotes(List<AtomicNoteEntity> atomicNoteEntity);

    @Query("DELETE FROM atomic_note_entities WHERE note_id = :noteId")
    void deleteAtomicNote(Long noteId);
}

File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/AtomicNoteEntitiesDao.kt
package com.originb.inkwisenote2.modules.smartnotes.data

import androidx.room.*

@Dao
interface AtomicNoteEntitiesDao {
    @Query("SELECT * FROM atomic_note_entities WHERE note_id = :noteId")
    fun getAtomicNote(noteId: Long): AtomicNoteEntity

    @get:Query("SELECT * FROM atomic_note_entities")
    val allAtomicNotes: MutableList<AtomicNoteEntity>

    @Query("SELECT * FROM atomic_note_entities WHERE note_id IN (:noteIds)")
    fun getAtomicNotes(noteIds: MutableSet<Long>): MutableList<AtomicNoteEntity>

    @Insert(onConflict = OnConflictStrategy.Companion.REPLACE)
    fun insertAtomicNote(atomicNoteEntity: AtomicNoteEntity): Long

    @Update
    fun updateAtomicNote(atomicNoteEntity: AtomicNoteEntity): Int

    @Update
    fun updateAtomicNotes(atomicNoteEntity: MutableList<AtomicNoteEntity>): Int

    @Query("DELETE FROM atomic_note_entities WHERE note_id = :noteId")
    fun deleteAtomicNote(noteId: Long)
}
